File: Pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, string argv[])                       //argv[0] = ./caesar, argv[1] = k
{
    if(argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }

    int k;
    k = atoi(argv[1]);
    while(k>26)
    {
        k = k - 26;
    }

    string plaintext;
    int ciphertext;
    printf("plaintext: ");
    plaintext = get_string();
    printf("ciphertext: ");

    for(int i = 0; i < strlen(plaintext); i++)
    {
       do
        {
            if(isalpha(plaintext[i]))
            {
                if(islower(plaintext[i]))
                {
                    ciphertext = (int) plaintext[i] + k;

                    if (ciphertext > 122)
                    {
                        ciphertext = ciphertext - 26;
                    }

                    printf("%c", (char) ciphertext);
                }

                else if(isupper(plaintext[i]))
                {
                    ciphertext = (int) plaintext[i] + k;

                    if (ciphertext > 90)
                    {
                        ciphertext = ciphertext - 26;
                    }
                    printf("%c", (char) ciphertext);
                }
            }

           else
            {
                printf("%c", plaintext[i]);
            }
        }
        while (plaintext[i] == '\0');
    }

    printf("\n");
}

File: Pset6/caesar.py
#chr(97) returns the string 'a'
#ord('a') returns the integer 97

import cs50
import sys

if len(sys.argv) != 2:
   print("Usage: python caesar.py k")
   exit(1)

k = int(sys.argv[1])

while k>26:
   k = k -26

plaintext = input("plaintext: ")
print("ciphertext: ", end="")

for char in plaintext:
   if char.isalpha():
       if char.islower():
           ciphertext = ord(char) + k
           if ciphertext > 122:
               ciphertext = ciphertext - 26
           print(chr(ciphertext), end="")
       elif char.isupper():
           ciphertext = ord(char) + k
           if ciphertext > 90:
              ciphertext = ciphertext - 26
           print(chr(ciphertext), end="")
   else:
       print(char, end="")
print()
